### PR TITLE
pp_reverse: don't COW buffer just to then un-COW it

### PR DESCRIPTION
**Pull request by richardleach**
When it has a single argument, scaler-mode `pp_reverse` copies the source PV buffer to the destination PV buffer via the `sv_setsv()` macro. This expands to:

        sv_setsv_flags(dsv, ssv, SV_GMAGIC|SV_DO_COW_SVSETSV)

The `SV_DO_COW_SVSETSV` flag means that Perl_sv_setsv_flags will try to achieve the "copy" via these mechanisms:
* Swipe the source buffer (if possible)
* COW the buffer (if possible)
* Copy the buffer

However, using COW makes no sense because pp_reverse is going to then modify the buffer. Almost immediately after the `sv_setsv` call, any COWing will be reversed by the call to `SvPV_force` and a full copy of the buffer taken. So this commit expands the `sv_setsv` call and drops the COW flag.

---------------------------------------------------------------------------------
* This set of changes does not require a perldelta entry.

### CODE REVIEW

**Comment by richardleach** [comment]
PS Doing a full copy also seems like not the best idea, since it read-writes the entire string (albeit between buffers), only for the function to then read-write the whole buffer at least once.

I have a branch in progress to separate out the "swipe" case from the "copy" case, where in the "copy" case the buffer isn't actually copied wholesale, instead the reverse operation reads from the source buffer and writes to the destination buffer.

Seems worth making this PR in case the WIP branch never lands.

File: pp.c [hunk 1]
```diff
@@ -6517,10 +6517,10 @@ PP_wrapped(pp_reverse, 0, 1)
             SP = MARK + 1;
```
Comment (tonycoz) [inline pp.c:6517]: "scaler" -> "scalar" in the commit message

**Comment by tonycoz** [comment]
LGTM except for the "scaler" in the commit message.

File: pp.c (hunk 1 above)
Comment (richardleach) [reply pp.c:6517]: Thanks, Tony

**Pull request merged**